packages/gradient-block/src/GradientBlock.tsx

```diff
@@ -0,0 +1,154 @@
+import { Divider } from '@frontify/fondue';
+import 'tailwindcss/tailwind.css';
+import { GradientColor, Settings, gradientHeightValues, gradientOrientationValues } from './types';
+import { DEFAULT_GRADIENT_COLORS, DEFAULT_HEIGHT_VALUE, DEFAULT_ORIENTATION_VALUE } from './constants';
+import { AddColorButton, ColorPicker, ColorTooltip, CssValueDisplay, SquareBadgesRow } from './components';
+import { toHexString } from '@frontify/guideline-blocks-shared';
+
+export const GradientBlock = ({ appBridge }: BlockProps) => {
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const isEditing = useEditorState(appBridge);
+    const {
+        gradientColors,
+        isOrientationCustom,
+        orientationCustom,
+        orientationSimple,
+        isHeightCustom,
+        heightCustom,
+        heightSimple,
+        displayCss,
+    } = blockSettings;
+    const gradientBlockRef = useRef<HTMLDivElement>(null);
+    const [addButtonPositionLeft, setAddButtonPositionLeft] = useState(0);
+    const [currentlyEditingPosition, setCurrentlyEditingPosition] = useState(0);
+    const [showAddButton, setShowAddButton] = useState(false);
+    const [showColorModal, setShowColorModal] = useState(false);
+    const [isRefReady, setRefReady] = useState(false);
+
+    if (!gradientColors) {
+        setBlockSettings({
+            gradientColors: DEFAULT_GRADIENT_COLORS,
+        });
+    }
+
+    const setGradientColors = (colors: GradientColor[]) => {
+        setBlockSettings({
+            gradientColors: colors,
+        });
+    };
+
+    const gradientOrientation = isOrientationCustom
+        ? orientationCustom
+        : gradientOrientationValues[orientationSimple ?? DEFAULT_ORIENTATION_VALUE];
+
+    const gradientBlockHeight = isHeightCustom
+        ? heightCustom
+        : gradientHeightValues[heightSimple ?? DEFAULT_HEIGHT_VALUE];
+
+    const parseGradientColorsToCss = () => {
+        if (!gradientColors) {
+            return '';
+        } else {
+            let colorsAsString = '';
+            for (const color of gradientColors.sort((a, b) => a.position - b.position)) {
+                colorsAsString += `, ${toHexString(color.color)} ${color.position}%`;
+            }
+
+            return `linear-gradient(${gradientOrientation}deg${colorsAsString})`;
+        }
+    };
+
+    const handleMouseMove = (event: MouseEvent<HTMLDivElement>) => {
+        if (showColorModal) {
+            return;
+        }
+        const rect = event.currentTarget.getBoundingClientRect();
+        const { clientX } = event.nativeEvent;
+        const relativeMouseX = clientX - rect.x;
+        setAddButtonPositionLeft(relativeMouseX - 16 / 2);
+        setShowAddButton(true);
+    };
+
+    useEffect(() => {
+        if (gradientBlockRef.current) {
+            setRefReady(true);
+        } else {
+            setRefReady(false);
+        }
+    }, [gradientBlockRef]);
```

Comment:
AFAIK putting refs in the dependency array has no effect. However once the useeffect hook runs you can be sure that the tree has been rendered at least once, so you can do something like `useEffect(()=>{ setRefReady(true) },[])`. Maybe a better name would be `isMounted`.